index.html

```diff
@@ -397,6 +403,71 @@ <h3>
           </li>
         </ol>
       </section>
+      <section>
+        <h3>
+          Steps for when a user changes payment method
+        </h3>
+        <p class="note" data-lt="BasicCardRequest">
+          When the user switches from one card to another, these steps cause an
+          event to fire in the <cite data-cite="!payment-request">Payment
+          Request API</cite>. The algorithm presupposes that the user agent is
+          only presenting <a>supported cards</a> to the end user, by having
+          filtered out unsupported cards based on the initiating payment
+          request's <a>BasicCardRequest</a>'s <a>supportedNetworks</a> and
+          <a>supportedTypes</a> values.
+        </p>
+        <p>
+          The <dfn>steps for when a user changes payment method</dfn> are as
+          follows:
+        </p>
+        <ol>
+          <li>From the <a>supported cards</a> presented to the end-user, let
+          <var>card</var> be the <a>card</a> the user selected.
+          </li>
+          <li>Let <var>methodDetails</var> be a newly created
+          <a>BasicCardChangeDetails</a> dictionary.
+          </li>
+          <li>Set <var>methodDetails</var>["<a>type</a>"] to the
+          <a>BasicCardType</a> that represents <var>card</var>'s <a>type</a>.
+          </li>
+          <li>If the <var>card</var> is part of a <a>network</a>, set
+          <var>methodDetails</var>["<a>network</a>"] to the network identifier
+          that represents <var>card</var>'s <a>network</a>
```

Comment:
I concur with @mnoorenberghe. As @aestes noted when opening https://github.com/w3c/payment-request/issues/662, the `applepaypaymentmethodchanged` event "allows merchants to add additional display items based on the type of card selected by the user." I'm not sure what these display items are, but presumably they are not new items in the cart but actually payment modifiers (e.g., take $5 off if you pay with debit or 10% if you use your store card). If so, a `PaymentDetailsModifier` sequence can be provided in the initial payment request and the user agent can apply the modifiers locally if the user selects a card of a different type. Does the user agent _really_ need to fire an event off to the merchant here (e.g., so that the merchant can dynamically determine what modifiers will be applied)? That feels a bit far-fetched and not worth the privacy leak.

Reply:
@mnoorenberghe, wrote: 
>  If the merchant wants to do anything meaningful with this data (e.g. to change the total and replace the need for modifiers) then the UA would have to dispatch this event as soon as the PaymentRequest opens and that would leak the user's default-selected type and network to the merchant. 

We discussed this at the F2F also. The solution is to apply the modifiers on `show()`, then no event is fired on the default card - the modified gets applied,  so no leakage occurs immediately. 

When the user then changes card, by actually clicking on it, then we would fire the event. 

I'll clarify that the event should only fire on explicit user action. 

> It sounds like Apple didn't send the network with their applepaypaymentmethodchanged event either so I'm not sure the network is needed. 

I was reaching the same conclusion... why I started `null`'ing it out :) I'll remove it. 

> Because of the increased fingerprinting potential (type + network) from a user-interaction showing a PR without the user completing it, I no longer like the idea of the event, especially if the network is leaked (since it provides more entropy). 
 
With the removal of network, and not firing on default card, I think the above is sufficient to mitigate the privacy risks, no?  

(note also that you can `.canMakePayment()` this information already, but we can deal with that separately.) 

@stpeter wrote:
> I'm not sure what these display items are, but presumably they are not new items in the cart but actually payment modifiers

The idea is that if you know the card type, you don't need the modifiers on subsequent changes. So, instead of having a modifier, the developer just updates the display items directly.